libraries/shared/factories/storage/README.md

```diff
@@ -1,89 +1,98 @@
 	metadata, lookupErr := transformer.StorageKeysLookup.Lookup(diff.StorageKey)
 	if lookupErr != nil {
-		return lookupErr
+		return fmt.Errorf("error getting metadata for storage key: %w", lookupErr)
 	}
-	value, decodeErr := utils.Decode(diff, metadata)
-	if decodeErr != nil {
-		return decodeErr
-	}
-	return transformer.Repository.Create(diff.BlockHeight, diff.BlockHash.Hex(), metadata, value)
+	value := storage.Decode(diff, metadata)
+	return transformer.Repository.Create(diff.ID, diff.HeaderID, metadata, value)
 }
 ```
 
+
 ## Custom Code
 
 In order to watch an additional smart contract, a developer must create three things:
 
-1. StorageKeysLoader - identify keys in the contract's storage trie, providing metadata to describe how associated values should be decoded.
+1. KeysLoader - identify keys in the contract's storage trie, providing metadata to describe how associated values should be decoded.
 1. Repository - specify how to persist a parsed version of the storage value matching the recognized storage key.
 1. Instance - create an instance of the storage transformer that uses your mappings and repository.
 
-### StorageKeysLoader
+### KeysLoader
 
-A `StorageKeysLoader` is used by the `StorageKeysLookup` object on a storage transformer.
+A `KeysLoader` is used by the `StorageKeysLookup` object on a storage transformer. You'll implement this interface.
 
-```golang
+```go
 type KeysLoader interface {
-	LoadMappings() (map[common.Hash]utils.StorageValueMetadata, error)
+	LoadMappings() (map[common.Hash]types.ValueMetadata, error)
 	SetDB(db *postgres.DB)
 }
 ```
 
 When a key is not found, the lookup object refreshes its known keys by calling the loader.
 
-```golang
+```go
 func (lookup *keysLookup) refreshMappings() error {
-	var err error
-	lookup.mappings, err = lookup.loader.LoadMappings()
+	newMappings, err := lookup.loader.LoadMappings()
 	if err != nil {
-		return err
+		return fmt.Errorf("error loading mappings: %w", err)
 	}
+	lookup.mappings = newMappings
 	return nil
 }
 ```
 
-A contract-specific implementation of the loader enables the storage transformer to fetch metadata associated with a storage key.
+A contract-specific implementation of the loader enables the storage transformer to fetch metadata associated with a storage key. Here is the version for the `flap` contract.
 
-Storage metadata contains: the name of the variable matching the storage key, a raw version of any keys associated with the variable (if the variable is a mapping), and the variable's type.
+``` go
+func (loader *keysLoader) LoadMappings() (map[common.Hash]types.ValueMetadata, error) {
+	mappings := loadStaticKeys()
+	mappings, wardsErr := loader.addWardsKeys(mappings)
+	if wardsErr != nil {
+		return nil, fmt.Errorf("error adding wards keys to flap keys loader: %w", wardsErr)
+	}
+	mappings, bidErr := loader.loadBidKeys(mappings)
+	if bidErr != nil {
+		return nil, fmt.Errorf("error adding bid keys to flap keys loader: %w", bidErr)
```

Comment:
I don't know why these ones have the `%` highlighted in red 🤔 

Used to seeing something like that when something isn't closed (parens, quotes, etc), but I'm not seeing anything missing

Reply:
That's weird - but they look okay rendered. Something with github's diffing algorithm?